frontends/web/src/routes/exchanges/exchanges.tsx

```diff
@@ -0,0 +1,237 @@
+/**
+ * Copyright 2020 Shift Devices AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, h, RenderableProps } from 'preact';
+import A from '../../components/anchor/anchor';
+import Button from '../../components/forms/button';
+import { SwissMadeOpenSource } from '../../components/icon/logo';
+import { Footer, Header } from '../../components/layout';
+import { Spinner } from '../../components/spinner/Spinner';
+import { translate, TranslateProps } from '../../decorators/translate';
+
+import * as styles from './exchanges.css';
+
+interface ExchangesProps {
+}
+
+type Props = ExchangesProps & TranslateProps;
+
+type Method = 'BT' | 'DCA' | 'CC' | 'SW' | 'P2P';
+type Region = 'NA' | 'LA' | 'EU' | 'AF' | 'APAC';
+
+interface State {
+    status: 'loading' | 'loaded' | 'error';
+    region: Region | null;
+    methods: Method[];
+}
+
+interface Exchange {
+    key: string;
+    link: string;
+    description: string;
+    regions: Region[];
+    payment: Method[];
+    hostname?: string;
+}
+
+class Exchanges extends Component<Props, State> {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            status: 'loading',
+            region: null,
+            methods: [],
+        };
+        this.data = [];
+    }
+
+    private data: Exchange[];
+
+    public componentDidMount() {
+        setTimeout(() => {
+            fetch('/assets/exchanges/exchanges.json')
+                .then(res => res.json())
+                .then(data => data.map(({link, ...rest}) => ({
+                    ...rest,
+                    link,
+                    hostname: new URL(link).hostname,
+                })))
+                .then(data => this.data = data)
+                .then(() => this.setState({ status: 'loaded', region: null, methods: [] }))
+                .catch(() => this.setState({ status: 'error' }));
+        }, 400);
```

Comment:
Oh, I see it now. Why wait 400ms? Wouldn't directly `fetch(...).then().then()` work otherwise. I mean, without `setTimeout`.

Reply:
The reason for it was that the response was too fast and you'd never see the Spinner animation. I will remove it :)

Reply:
fixed in 6f66ac31f4aba1cf872d83154479920be933cfff